### PR TITLE
HIVE-26997: Iceberg: Vectorization gets disabled at runtime in merge-into statement

### PR DESCRIPTION
**Pull request by deniskuzZ**
### What changes were proposed in this pull request?

Fixed non-vectorization cause

### Why are the changes needed?

Performance

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

q test

### CODE REVIEW

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergAcidUtil.java [hunk 1]
```diff
@@ -93,10 +95,16 @@ public static Schema createFileReadSchemaWithVirtualColums(List<Types.NestedFiel
...
    * @return The schema for SerDe operations, extended with metadata columns needed for deletes
    */
-  public static Schema createSerdeSchemaForDelete(List<Types.NestedField> dataCols) {
-    List<Types.NestedField> cols = Lists.newArrayListWithCapacity(dataCols.size() + SERDE_META_COLS.size());
+  public static Schema createSerdeSchemaForDelete(List<Types.NestedField> dataCols, boolean partitioned,
+      Properties serDeProperties) {
+    boolean skipRowData = Boolean.parseBoolean(serDeProperties.getProperty(WriterBuilder.ICEBERG_DELETE_SKIPROWDATA,
+        WriterBuilder.ICEBERG_DELETE_SKIPROWDATA_DEFAULT));
+    List<Types.NestedField> cols = Lists.newArrayListWithCapacity(
+        SERDE_META_COLS.size() + (skipRowData || partitioned ? 0 : dataCols.size()));
```
Comment (kasakrisz) [inline iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergAcidUtil.java:103]: is it `skipRowData && !partitioned` ?

File: iceberg/iceberg-handler/src/test/queries/positive/vectorized_iceberg_merge_mixed.q [hunk 2]
```diff
@@ -0,0 +1,197 @@
+-- SORT_QUERY_RESULTS
```
Comment (kasakrisz) [inline iceberg/iceberg-handler/src/test/queries/positive/vectorized_iceberg_merge_mixed.q:1]: Please add test for merging partitioned iceberg tables.

**Review by kasakrisz — CHANGES_REQUESTED** [review]

**Review by kasakrisz — APPROVED** [review]
LGTM, pending test

File: ql/src/java/org/apache/hadoop/hive/ql/parse/RewriteSemanticAnalyzer.java [hunk 3]
```diff
@@ -670,19 +670,25 @@ public List<String> getSortKeys() {
   }
 
   protected static class NonNativeAcidColumnAppender extends ColumnAppender {
+    private String deletePrefix;
```
[the remark this answers is not in the document]
Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/RewriteSemanticAnalyzer.java:673]: yep

**Pull request merged**